app/_src/mesh/features/openshift-quickstart.md

```diff
@@ -0,0 +1,184 @@
+
+* How to use the Red Hat Certified Kong Mesh Images
+* How to implement the required OpenShift SecurityContextConstraints (SCC) for the kong-mesh sidecar
+* Deploy and join the Kong for Kubernetes Ingress Controller (KIC) to the mesh
+* Deploy a sample application, bookinfo, on the mesh and validate it's all working
+
+## Prerequisites
+* [ROSA CLI installed](https://docs.openshift.com/rosa/rosa_install_access_delete_clusters/rosa_getting_started_iam/rosa-installing-rosa.html) or another OpenShift 4.12 cluster with the ability to create LoadBalancer type Kubernetes Services
+* [kubectl CLI installed](https://kubernetes.io/docs/tasks/tools/)
+* [OpenShift CLI installed](https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/getting-started-cli.html)
+* [Helm 3 installed](https://helm.sh/docs/intro/install/)
+
+## Install ROSA
+
+1. Create System Variables:
+    ```bash
+    CLUSTER_NAME=kong-mesh-2
+    REGION=us-west-2
+    ```
+
+1. Create a small ROSA cluster:
+    ```bash
+    rosa create cluster --cluster-name=$CLUSTER_NAME --region=$REGION --multi-az=false --version 4.12.13
+    ```
+
+1. When the Cluster install is complete, create a cluster-admin user:
+    ```bash
+    rosa create admin --cluster $CLUSTER_NAME
+    ```
+
+1. Validate you can login to the cluster via the credentials provided by the rosa cli stdout. Once login is successful you can proceed to the next step.
+
+## Install Kong Mesh in Standalone Mode
+
+1. Create the `kong-mesh-system` namespace:
+    ```bash
+    kubectl create namespace kong-mesh-system
+    ```
+
+1. Create the image pull secret:
+    ```bash
+    kubectl create secret docker-registry rh-registry-secret -n kong-mesh-system \
+        --docker-server=registry.connect.redhat.com \
+        --docker-username=<username> \
+        --docker-password=<password> \
+        --docker-email=<email>
+    ```
+
+1. Add nonroot-v2 to job service accounts:
+    ```bash
+    oc adm policy add-scc-to-user nonroot-v2 system:serviceaccount:kong-mesh-system:kong-mesh-install-crds
+    oc adm policy add-scc-to-user nonroot-v2 system:serviceaccount:kong-mesh-system:kong-mesh-patch-ns-job 
+    oc adm policy add-scc-to-user nonroot-v2 system:serviceaccount:kong-mesh-system:kong-mesh-pre-delete-job
+    ```
+
+1. Grab the latest helm chart:
+    ```bash
+    helm repo add kong-mesh https://kong.github.io/kong-mesh-charts
+    ```
+
+1. Install Kong Mesh:
+    ```bash
+    helm upgrade -i kong-mesh kong-mesh/kong-mesh --version 2.2.0 -f kong-mesh/values.yaml -n kong-mesh-system
+    ```
+
+1. Open a second terminal and port-forward to reach the mesh GUI:
+    ```bash
+    kubectl port-forward svc/kong-mesh-control-plane -n kong-mesh-system 5681:5681
+    ```
+
+    You should be able to reach the Kong-Mesh UI at `http://localhost:5681/gui`.
+
+
+1. Last, do some prep work for the sidecar itself so sidecars will startup successfully. Apply the kong-mesh-sidecar scc and corresponding container patches:
```

Comment:
How does this prep them to start successfully?

Reply:
These are two new custom files we need to bring into the doc or host somewhere. 

1. [kong-mesh-sidecar-scc](https://github.com/Kong/kong-mesh-quickstart-openshift/blob/main/kong-mesh/kong-mesh-sidecar-scc.yaml) : 
2. [container-patch](https://github.com/Kong/kong-mesh-quickstart-openshift/blob/main/kong-mesh/container-patch.yaml)

In Openshift, by default, all pods and containers are given the restricted security context constraint (SCC) but this is insufficient for the mesh sidecars (containers). The sidecars need a slightly escalated permissions, these permissions are defined in the sidecar-scc manifest. 

Step 1 - create a custom SCC, but this manifest is not associated to any pod container or sidecar in Openshift. 

Step 2 - defines how to patch the sidecars with the needed SCC. 

There is a third step. Step 3 - line 133 - all application pods that will have sidecars, need to have their service accounts given permission to use the custom SCC.